src/scout_apm/falcon.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from scout_apm.api import install
+from scout_apm.core.ignore import ignore_path
+from scout_apm.core.tracked_request import TrackedRequest
+
+try:
+    import falcon
+except ImportError:  # pragma: no cover
+    falcon = None
+
+# Falcon Middleware docs:
+# https://falcon.readthedocs.io/en/stable/api/middleware.html
+
+
+class ScoutMiddleware(object):
+    """
+    Falcon Middleware for integration with Scout APMi.
+    """
+
+    def __init__(self, config):
+        install(config=config)
+        self.api = None
+
+    def set_api(self, api):
+        if not isinstance(api, falcon.API):
+            raise ValueError("api should be an instance of falcon.API")
+        self.api = api
+
+    def process_request(self, req, resp):
+        if self.api is None:
+            raise RuntimeError(
+                "Call {}.set_api() before requests begin".format(
+                    self.__class__.__name__
+                )
+            )
+
+        tracked_request = TrackedRequest.instance()
+        tracked_request.mark_real_request()
+        req.context.scout_tracked_request = tracked_request
```

Comment:
Is there a possibility that `req.context` and `TrackedRequest.instance()` return a different TrackedRequest?

Reply:
In adverse circumstances where some other code (i.e. a user integration with the API) finishes the request, yes. I think this way will die more reliably though in such adverse cases - the other integration that always goes through `TrackedRequest.instance()` can split data across separate TrackedRequests which might be intended for different URL's. With this approach, we could just see double tracking of some requests as they will `finish()` twice.